src/js/GlobalSettings/GlobalSettingsProvider.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from "react";
+import EthereumProvider from "../components/EthereumProvider/EthereumProvider";
+import AccountProvider from "../components/AccountProvider/AccountProvider";
+import {
+  ConnectWay,
+  Hardware,
+  IGlobalSettingsProvider,
+  IProviderProps,
+} from "../types/Types";
+import AlertDialogErrorProvider from "../components/AlertDialogErrorProvider/AlertDialogErrorProvider";
+import MetamaskProvider from "../components/ConnectMetamask/MetamaskProvider";
+import {
+  DEFAULT_HARDWARE_TYPE,
+  detectHardware,
+  isPhoneHardware,
+} from "./Hardware/Hardware";
+import useMainContent from "./MainContentView/useMainContent";
+
+const DEFAULT_CONTEXT_VALUE: IGlobalSettingsProvider = {
+  hardware: DEFAULT_HARDWARE_TYPE,
+  isPhoneHardware,
+  mainContent: "wallet",
+  setMainContent: () => {},
+  connectWay: { current: "" },
+};
+
+export const GlobalSettingsContext =
+  React.createContext<IGlobalSettingsProvider>(DEFAULT_CONTEXT_VALUE);
+
+export default function GlobalSettingsProvider({
+  children,
+}: IProviderProps): JSX.Element {
+  const { mainContent, setMainContent } = useMainContent();
+  const connectWay = React.useRef<ConnectWay>("");
+
+  const hardware: Hardware = React.useMemo(() => detectHardware(), []);
+
+  const contextValue = React.useMemo(() => {
+    return {
+      hardware,
+      isPhoneHardware,
```

Comment:
Arrow functions have the shortcut view for return value
```
  const contextValue = React.useMemo(() => ({
    hardware,
    ...
```

Reply:
Done